src/main/java/org/openstreetmap/atlas/checks/validation/tag/SimilarTagValueCheck.java

```diff
@@ -0,0 +1,346 @@
+            List.of("formal", "informal"), List.of("hotel", "hostel"), List.of("hump", "bump"),
+            List.of("seed", "feed"));
+
+    private final List<String> tagsToIgnore;
+    private final List<String> tagsWithSubCategoriesToIgnore;
+    private final List<List<String>> commonSimilars;
+    private final Double minValueLength;
+    private final Double minSimilarityThreshold;
+    private final Double maxSimilarityThreshold;
+
+    /**
+     * @param configuration
+     *            the JSON configuration for this check
+     */
+    public SimilarTagValueCheck(final Configuration configuration)
+    {
+        super(configuration);
+        this.commonSimilars = this.configurationValue(configuration, "filter.commonSimilars",
+                COMMON_SIMILARS_DEFAULT);
+        this.tagsToIgnore = this.configurationValue(configuration, "filter.tags",
+                TAGS_TO_IGNORE_DEFAULT);
+        this.tagsWithSubCategoriesToIgnore = this.configurationValue(configuration,
+                "filter.tagsWithSubCategories", TAGS_WITH_SUB_CATEGORIES_TO_IGNORE_DEFAULT);
+        this.minValueLength = this.configurationValue(configuration, "value.length.min",
+                MIN_VALUE_LENGTH, Double::doubleValue);
+        this.minSimilarityThreshold = this.configurationValue(configuration,
+                "similarity.threshold.min", MIN_SIMILARITY_THRESHOLD_DEFAULT, Double::doubleValue);
+        this.maxSimilarityThreshold = this.configurationValue(configuration,
+                "similarity.threshold.max", MAX_SIMILARITY_THRESHOLD_DEFAULT, Double::doubleValue);
+    }
+
+    /**
+     * Valid objects for this check objects with tags that contain multiple values.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return {@code true} if this object should be checked
+     */
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        // Only want objects with tags that contain multiple values.
+        return object.getOsmTags().values().stream().anyMatch(value -> value.contains(SEMICOLON));
+    }
+
+    /**
+     * This is the actual function that will check to see whether the object needs to be flagged.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return an optional {@link CheckFlag} object
+     */
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        final Map<String, String> tagsWithMultipleValues = object.getOsmTags().entrySet().stream()
+                .filter(entry -> !this.tagsToIgnore.contains(entry.getKey()))
+                .filter(Predicate.not(this::isTagWithSubCategoriesToIgnore))
+                .map(this::removeCommonFalsePositiveValues)
+                // Only keep tags that still have multiple values
+                .filter(entry -> entry.getValue().contains(SEMICOLON))
```

Comment:
could be here too if you prefer, to drive down the # of expensive regex checks

Reply:
When a tag gets to that the last filter its value will not have a semicolon as it's last character. While it wasn't something I was considering when I wrote it, any tag with a semicolon as its last character will get it removed in the 'removeCommonFalsePositiveValues'.